lib/core/board.rb

```diff
@@ -1,30 +1,53 @@
 class Board
-  UNCHECKED_TILE = ' '
+  UNCHECKED_TILE = " "
 
   attr_reader :size
+  attr_accessor :board
 
   def initialize(size)
     @size = size
-    @simplified_board = Array.new(size*size, UNCHECKED_TILE)
+    @board = Array.new(size*size, UNCHECKED_TILE)
+  end
+
+  def self.create_from_array(array)
+    instance = Board.new(3)
+    instance.board = array.clone
+    instance
+  end
+
+  def clone
+    instance = Board.new(@size)
+    instance.board = board.clone
+    instance
+  end
+
+  def clone_with_new_state(move, player)
+    instance = self.clone
+    instance.mark_tile(move, player)
+    instance
+  end
+
+  def total_tiles
+    @size * @size
   end
 
   def tile(position)
-    @simplified_board[position]
+    @board[position]
   end
 
   def row(index)
-    @simplified_board.each_slice(3).to_a[index]
+    @board.each_slice(3).to_a[index]
   end
 
   def column(index)
-    @simplified_board.each_slice(3).to_a.transpose[index]
+    @board.each_slice(3).to_a.transpose[index]
   end
 
   def diagonal
     tiles = []
     position = 0
-    until position > (size*size - 1) # In a 3x3 board, that would be 8 due to 0-based indexes.
-      tiles << @simplified_board[position]
+    until position > (total_tiles - 1) # In a 3x3 board, that would be 8 due to 0-based indexes.
```

Comment:
Any time I have a conditional with somewhat complex logic and a comment, I'd extract that into a method with a good name to remove the need for the comment. Make your code expressive such that comments aren't needed (when possible).